### PR TITLE
Expose bug with default constructors

### PR DESCRIPTION
**Pull request by cmelchior**
Apparently the recent change to getters/setters (https://github.com/realm/realm-java/pull/4206)  caused a regression not caught by our unit tests, but it was revealed by our Kotlin example (we should really re-enable checking those again).

/cc @zaki50

### CODE REVIEW

**Comment by cmelchior** [comment]
Note this is blocking our 2.3.2 release, so we need to either fix it fast or revert the offending change.

**Comment by cmelchior** [comment]
Instead of only injecting the context in the default constructor I changed it to injecting it in every constructor.

**Comment by zaki50** [comment]
@cmelchior Yes. We need to inject from all constructors.
I'll review this.

File: realm/realm-library/src/androidTest/java/io/realm/entities/DefaultValueFromOtherConstructor.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 Realm Inc.
```
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/entities/DefaultValueFromOtherConstructor.java]: 2017

File: realm/realm-library/src/androidTest/java/io/realm/entities/DefaultValueFromOtherConstructor.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+package io.realm.entities;
+
+import java.util.Date;
+import java.util.UUID;
+
+import io.realm.RealmList;
+import io.realm.RealmObject;
+import io.realm.annotations.Ignore;
+import io.realm.annotations.PrimaryKey;
```
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/entities/DefaultValueFromOtherConstructor.java]: `RealmList` and `PrimaryKey` is unused.

**Pull request merged**